### PR TITLE
Make the js-lib to work with maas mobile flow

### PR DESCRIPTION
**Pull request by simmol**
The library now react on received data from the server ( based on status) and propagete when status
changes to the client, so appropriate message can be shown to the user

### CODE REVIEW

File: lib/mpin.js [hunk 1]
```diff
@@ -616,14 +616,15 @@ var mpinjs = (function () {
     });
   };
 
-  Mpin.prototype.waitForMobileAuth = function (timeoutSeconds, requestSeconds, cb) {
+  Mpin.prototype.waitForMobileAuth = function (timeoutSeconds, requestSeconds, cb, cb2) {
```
Comment (simeon-aladjem) [inline lib/mpin.js]: `cb2` is a bad name, lets give it a meaningful name

File: lib/mpin.js [hunk 2]
```diff
@@ -616,14 +616,15 @@ var mpinjs = (function () {
...
     var self = this, _reqData = {};
     if (!this.webOTT) {
       return cb({code: Errors.wrongFlow.code, type: Errors.wrongFlow.type, message: "Need to call getAccessNumber method before this."}, null);
     } else if (!timeoutSeconds) {
       return cb({code: Errors.missingParams.code, type: Errors.missingParams.type, message: "Missing timeout/expiration period(in seconds)."}, null);
     }
 
+    self.mobile_status = self.mobile_status || '';
```
Comment (simeon-aladjem) [inline lib/mpin.js]: Here also the naming convention is not followed - it would be better to name this `mobileStatus`

**Comment by simeon-aladjem** [comment]
LGTM

**Comment by bbvox** [comment]
OK

**Pull request merged**